proposed/0016-launcher-data_types-api/0016-launcher-data-types-api.md

```diff
@@ -0,0 +1,282 @@
+- Feature Name: Expose API for structured data and immutable data from safe_launcher
+- Type: new feature
+- Related components: safe_launcher
+- Start Date: 20-11-2015
+- RFC PR:
+- Issue number:
+
+# Summary
+
+Expose API from launcher to perform operations on Structured data and
+Immutable data through the launcher API.
+
+# Motivation
+
+Structured data is designed in such a way to enable the developers to create their own
+data structures.
+
+For example, safe_nfs uses Structured data for storing the DirectoryListing in the SafeNetwork.
+safe_dns uses the logic as detailed here in this [RFC](https://github.com/maidsafe/rfcs/blob/master/implemented/0002-name-service/0002-name-service.md#detailed-design) to save the name service in the SafeNetwork.
+
+Providing these APIs would open up the possibilities of using the structured
+data to fit the needs of the developers to manage their own data structures,
+which would facilitate in building wide range of applications on SafeNetwork.
+
+Structured data is detailed in this implemented [unified structured data RFC](https://github.com/maidsafe/rfcs/blob/master/implemented/0000-Unified-structured-data/0000-Unified-structured-data.md#structured-data).
+
+# Detailed design
+
+The APIs that would facilitate operations on the SafeNetwork Data types can be grouped
+under the end point `safe-api/v1.0/data_types/`.
+
+## Structured data API
+
+#### Create Structured data
+Structure Data can be created only within the permitted range as specified in the [reserved-names RFC](https://github.com/maidsafe/rfcs/blob/master/implemented/0003-reserved_names/0003-reserved_names.md#detailed-design),
+i.e within the permissible range 10,001 to 2^64
```

Comment:
This presumably refers to the tag_type, but it is confusing as the text refers to both "names" (ie text) and "range 10,001 to 2^64" (numeric) so it would help to be explicit.